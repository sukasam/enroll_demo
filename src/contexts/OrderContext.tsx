import { TransactionMethod } from "Services/Hydra/order/types";

import {
    ReactElement,
    ReactNode,
    createContext,
    useCallback,
    useContext,
    useEffect,
    useMemo,
    useState
} from "react";

import useOrderCalculations from "./hooks/OrderCalculations";
import { useProducts } from "./ProductContext";

import {
    AddressType,
    BankWireInfo,
    CashPaymentInfo,
    EventDetails,
    OrderContextType,
    OrderResult,
    PayPalInfo,
    PaymentMethod,
    ShippingMethodQuoteType
} from "./types/OrderContextTypes";

const defaultOrderContext: OrderContextType = {
    orderResult: null,
    setOrderResult: () => null,
    shippingMethodQuotes: [],
    setShippingMethodQuotes: () => [],
    selectedShippingMethod: null,
    setSelectedShippingMethod: () => null,
    selectedPaymentMethod: null,
    setSelectedPaymentMethod: () => null,
    shipToAddress: null,
    setShipToAddress: () => null,
    signedTerms: false,
    setSignedTerms: () => false,
    userSignature: null,
    setUserSignature: () => null,
    taxId: "",
    setTaxId: () => null,
    secondaryTaxId: "",
    setSecondaryTaxId: () => null,
    paymentMethod: null,
    setPaymentMethod: () => null,
    billingAddress: null,
    setBillingAddress: () => null,
    signatureDateTimeStamp: null,
    setSignatureDateTimeStamp: () => null,
    payPal: null,
    setPayPal: () => null,
    cashPayment: null,
    setCashPayment: () => null,
    bankWire: null,
    setBankWire: () => null,
    isShipToOffice: false,
    setIsShipToOffice: () => false,
    isCreditCardPaymentValid: false,
    setIsCreditCardPaymentValid: () => false,
    selectedWillCallAddress: null,
    setSelectedWillCallAddress: () => null,
    pickUpAddressKey: null,
    setPickUpAddressKey: () => null,
    tppPayload: null,
    setTppPayload: () => null,
    isDigitalOnlyOrder: false,
    feedbackSubmitted: false,
    setFeedbackSubmitted: () => false,
    isEnrollmentCompleted: false,
    setIsEnrollmentCompleted: () => false
};

const OrderContext = createContext<OrderContextType>(defaultOrderContext);
OrderContext.displayName = "Order Context";

export function OrderProvider({
    children,
    testValues
}: {
    children: ReactNode;
    testValues?: Record<string, string>;
}): ReactElement {
    const { shoppingCart } = useProducts();
    const [orderResult, setOrderResult] = useState<OrderResult | null>(
        defaultOrderContext.orderResult
    );
    const [shippingMethodQuotes, setShippingMethodQuotes] = useState<
        ShippingMethodQuoteType[]
    >(defaultOrderContext.shippingMethodQuotes);
    const [selectedShippingMethod, setSelectedShippingMethod] =
        useState<ShippingMethodQuoteType | null>(
            defaultOrderContext.selectedShippingMethod
        );
    const [selectedPaymentMethod, setSelectedPaymentMethod] =
        useState<TransactionMethod | null>(
            defaultOrderContext.selectedPaymentMethod
        );
    const [shipToAddress, setShipToAddress] = useState<AddressType | null>(
        defaultOrderContext.shipToAddress
    );
    const [signedTerms, setSignedTerms] = useState<boolean>(
        defaultOrderContext.signedTerms
    );
    const [userSignature, setUserSignature] = useState<string | null>(
        defaultOrderContext.userSignature
    );
    const [taxId, setTaxId] = useState<string>(defaultOrderContext.taxId);
    const [secondaryTaxId, setSecondaryTaxId] = useState<string>(
        defaultOrderContext.secondaryTaxId
    );
    const [paymentMethod, setPaymentMethod] = useState<PaymentMethod | null>(
        null
    );
    const [billingAddress, setBillingAddress] = useState<AddressType | null>(
        defaultOrderContext.billingAddress
    );
    const [signatureDateTimeStamp, setSignatureDateTimeStamp] =
        useState<Date | null>(defaultOrderContext.signatureDateTimeStamp);
    const [payPal, setPayPal] = useState<PayPalInfo | null>(
        defaultOrderContext.payPal
    );
    const [cashPayment, setCashPayment] = useState<CashPaymentInfo | null>(
        defaultOrderContext.cashPayment
    );
    const [bankWire, setBankWire] = useState<BankWireInfo | null>(
        defaultOrderContext.bankWire
    );
    const [isShipToOffice, setIsShipToOffice] = useState<boolean>(false);
    const [isCreditCardPaymentValid, setIsCreditCardPaymentValid] =
        useState<boolean>(defaultOrderContext.isCreditCardPaymentValid);
    const [selectedWillCallAddress, setSelectedWillCallAddress] = useState<
        string | null
    >(null);
    const [pickUpAddressKey, setPickUpAddressKey] = useState<string | null>(
        defaultOrderContext.pickUpAddressKey
    );
    const [tppPayload, setTppPayload] = useState<EventDetails | null>(null);

    const { isDigitalOnlyOrder } = useOrderCalculations(shoppingCart);

    const [feedbackSubmitted, setFeedbackSubmitted] = useState<boolean>(false);

    const [isEnrollmentCompleted, setIsEnrollmentCompleted] =
        useState<boolean>(false);

    const loadCachedOrderData = (): void => {
        const cachedData = sessionStorage.getItem("orderContext");
        if (cachedData) {
            const parsedData: OrderContextType = JSON.parse(cachedData);
            setOrderResult(parsedData.orderResult);
            setShippingMethodQuotes(parsedData.shippingMethodQuotes);
            setSelectedShippingMethod(parsedData.selectedShippingMethod);
            setSelectedPaymentMethod(parsedData.selectedPaymentMethod);
            setShipToAddress(parsedData.shipToAddress);
            setSignedTerms(parsedData.signedTerms);
            setUserSignature(parsedData.userSignature);
            setTaxId(parsedData.taxId);
            setSecondaryTaxId(parsedData.secondaryTaxId);
            setPaymentMethod(parsedData.paymentMethod);
            setBillingAddress(parsedData.billingAddress);
            setSignatureDateTimeStamp(parsedData.signatureDateTimeStamp);
            setPayPal(parsedData.payPal);
            setCashPayment(parsedData.cashPayment);
            setBankWire(parsedData.bankWire);
            setIsShipToOffice(parsedData.isShipToOffice);
            setIsCreditCardPaymentValid(parsedData.isCreditCardPaymentValid);
            setSelectedWillCallAddress(parsedData.selectedWillCallAddress);
            setPickUpAddressKey(parsedData.pickUpAddressKey);
            setTppPayload(parsedData.tppPayload);
            setFeedbackSubmitted(parsedData.feedbackSubmitted);
            setIsEnrollmentCompleted(parsedData.isEnrollmentCompleted);
        }
    };

    useEffect(() => {
        loadCachedOrderData();
    }, []);

    const updateCachedOrderData = useCallback((): void => {
        const dataToCache = {
            orderResult,
            shippingMethodQuotes,
            selectedShippingMethod,
            selectedPaymentMethod,
            shipToAddress,
            signedTerms,
            taxId,
            secondaryTaxId,
            paymentMethod,
            userSignature,
            billingAddress,
            signatureDateTimeStamp,
            payPal,
            cashPayment,
            bankWire,
            isShipToOffice,
            isCreditCardPaymentValid,
            selectedWillCallAddress,
            pickUpAddressKey,
            tppPayload,
            feedbackSubmitted,
            isEnrollmentCompleted
        };
        sessionStorage.setItem("orderContext", JSON.stringify(dataToCache));
    }, [
        orderResult,
        shippingMethodQuotes,
        selectedShippingMethod,
        selectedPaymentMethod,
        shipToAddress,
        signedTerms,
        taxId,
        secondaryTaxId,
        paymentMethod,
        userSignature,
        billingAddress,
        signatureDateTimeStamp,
        payPal,
        cashPayment,
        bankWire,
        isShipToOffice,
        isCreditCardPaymentValid,
        selectedWillCallAddress,
        pickUpAddressKey,
        tppPayload,
        feedbackSubmitted,
        isEnrollmentCompleted
    ]);

    useEffect(() => {
        updateCachedOrderData();
    }, [updateCachedOrderData]);

    const value = useMemo(
        () => ({
            orderResult,
            setOrderResult,
            shippingMethodQuotes,
            setShippingMethodQuotes,
            selectedShippingMethod,
            setSelectedShippingMethod,
            selectedPaymentMethod,
            setSelectedPaymentMethod,
            shipToAddress,
            setShipToAddress,
            signedTerms,
            setSignedTerms,
            taxId,
            setTaxId,
            secondaryTaxId,
            setSecondaryTaxId,
            paymentMethod,
            setPaymentMethod,
            userSignature,
            setUserSignature,
            billingAddress,
            setBillingAddress,
            signatureDateTimeStamp,
            setSignatureDateTimeStamp,
            payPal,
            setPayPal,
            cashPayment,
            setCashPayment,
            bankWire,
            setBankWire,
            isShipToOffice,
            setIsShipToOffice,
            isCreditCardPaymentValid,
            setIsCreditCardPaymentValid,
            selectedWillCallAddress,
            setSelectedWillCallAddress,
            pickUpAddressKey,
            setPickUpAddressKey,
            tppPayload,
            setTppPayload,
            feedbackSubmitted,
            setFeedbackSubmitted,
            isDigitalOnlyOrder,
            isEnrollmentCompleted,
            setIsEnrollmentCompleted
        }),
        [
            orderResult,
            shippingMethodQuotes,
            selectedShippingMethod,
            selectedPaymentMethod,
            shipToAddress,
            signedTerms,
            taxId,
            secondaryTaxId,
            paymentMethod,
            userSignature,
            billingAddress,
            signatureDateTimeStamp,
            payPal,
            cashPayment,
            bankWire,
            isShipToOffice,
            isCreditCardPaymentValid,
            selectedWillCallAddress,
            pickUpAddressKey,
            tppPayload,
            feedbackSubmitted,
            isDigitalOnlyOrder,
            isEnrollmentCompleted
        ]
    );

    const providerValue = useMemo(
        () => ({
            ...value,
            ...testValues
        }),
        [value, testValues]
    );

    return (
        <OrderContext.Provider value={providerValue}>
            {children}
        </OrderContext.Provider>
    );
}

export const useOrder = (): OrderContextType => useContext(OrderContext);
