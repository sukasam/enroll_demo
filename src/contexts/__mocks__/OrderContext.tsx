import { TransactionMethod } from "Services/Hydra/order/types";
import {
    AddressType,
    BankWireInfo,
    CashPaymentInfo,
    EventDetails,
    OrderResult,
    PayPalInfo,
    PaymentMethod,
    ShippingMethodQuoteType
} from "../types/OrderContextTypes";

export const useOrder = jest.fn().mockReturnValue({
    orderResult: null as OrderResult | null,
    setOrderResult: jest.fn(),
    shippingMethodQuotes: [] as ShippingMethodQuoteType[],
    setShippingMethodQuotes: jest.fn(),
    selectedShippingMethod: null as ShippingMethodQuoteType | null,
    setSelectedShippingMethod: jest.fn(),
    selectedPaymentMethod: null as TransactionMethod | null,
    setSelectedPaymentMethod: jest.fn(),
    shipToAddress: null as AddressType | null,
    setShipToAddress: jest.fn(),
    signedTerms: false,
    setSignedTerms: jest.fn(),
    userSignature: null as string | null,
    setUserSignature: jest.fn(),
    taxId: "",
    setTaxId: jest.fn(),
    secondaryTaxId: "",
    setSecondaryTaxId: jest.fn(),
    paymentMethod: null as PaymentMethod | null,
    setPaymentMethod: jest.fn(),
    billingAddress: null as AddressType | null,
    setBillingAddress: jest.fn(),
    signatureDateTimeStamp: null as Date | null,
    setSignatureDateTimeStamp: jest.fn(),
    payPal: null as PayPalInfo | null,
    setPayPal: jest.fn(),
    cashPayment: null as CashPaymentInfo | null,
    setCashPayment: jest.fn(),
    bankWire: null as BankWireInfo | null,
    setBankWire: jest.fn(),
    isShipToOffice: false,
    setIsShipToOffice: jest.fn(),
    isCreditCardPaymentValid: false,
    setIsCreditCardPaymentValid: jest.fn(),
    selectedWillCallAddress: null as string | null,
    setSelectedWillCallAddress: jest.fn(),
    pickUpAddressKey: null as string | null,
    setPickUpAddressKey: jest.fn(),
    tppPayload: null as EventDetails | null,
    setTppPayload: jest.fn(),
    isDigitalOnlyOrder: false,
    feedbackSubmitted: false,
    setFeedbackSubmitted: jest.fn(),
    isEnrollmentCompleted: false,
    setIsEnrollmentCompleted: jest.fn()
});

export default useOrder;
