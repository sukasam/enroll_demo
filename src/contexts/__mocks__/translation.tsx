import { Alpha2 } from "Constants/countryConfig/enums.js";

export const useTranslations = jest.fn().mockReturnValue({
    country: "US" as Alpha2,
    language: "en",
    setCountry: jest.fn(),
    setLanguage: jest.fn(),
    fetchTranslations: jest.fn(),
    translations: {}
});

export default useTranslations;
