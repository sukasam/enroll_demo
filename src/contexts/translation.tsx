import { debounce } from "@mui/material";
import { translationsSvc } from "Hydra/translationsSvc";
import { sendEvent } from "Services/googleAnalytics";
import { getCookie, setCookie } from "cookies-next";
import {
    ReactNode,
    createContext,
    useCallback,
    useContext,
    useEffect,
    useMemo,
    useState
} from "react";

export interface TranslationContextType {
    translations: Record<string, string>;
    basicTranslations: Record<string, string>;
    country: string;
    language: string;
    setTranslations: (translations: Record<string, string>) => void;
    setBasicTranslations: (basicTranslations: Record<string, string>) => void;
    setCountry: (country: string) => void;
    setLanguage: (language: string) => void;
    fetchTranslations: (country: string, language: string) => void;
}

const defaultState: TranslationContextType = {
    translations: {},
    basicTranslations: {},
    country: getCookie("country") || "US",
    language: getCookie("language") || "en",
    setTranslations: () => null,
    setBasicTranslations: () => null,
    setCountry: () => null,
    setLanguage: () => null,
    fetchTranslations: () => null
};

function getInitialState(): TranslationContextType {
    if (typeof window !== "undefined") {
        const cachedData = sessionStorage.getItem("translationContext");
        if (cachedData) {
            try {
                return JSON.parse(cachedData);
            } catch (error) {
                console.error(
                    "Failed to parse translation context data from sessionStorage:",
                    error
                );
            }
        }
    }
    return defaultState;
}

const TranslationContext = createContext<TranslationContextType>(defaultState);
TranslationContext.displayName = "Translations Context";

function HtmlLangSync(props: { language: string }): React.ReactElement | null {
    const { language } = props;
    useEffect(() => {
        if (typeof document !== "undefined" && language) {
            document.documentElement.lang = language;
        }
    }, [language]);
    return null;
}

export function TranslationProvider({
    children,
    testValues
}: {
    children: ReactNode;
    testValues?: Record<string, string>;
}): React.ReactElement {
    const initialState = useMemo(() => getInitialState(), []);

    const [translations, setTranslations] = useState<Record<string, string>>(
        initialState.translations
    );
    const [basicTranslations, setBasicTranslations] = useState<
        Record<string, string>
    >(initialState.basicTranslations);
    const [country, setCountry] = useState<string>(initialState.country);
    const [language, setLanguage] = useState<string>(initialState.language);

    useEffect(() => {
        setCookie("country", country);
        setCookie("language", language);
        sendEvent("page_configration", {
            country,
            language
        });
    }, [country, language]);

    const fetchTranslations = debounce(
        async (_country: string, _language: string) => {
            try {
                const response = await translationsSvc.get({
                    country: _country || country,
                    language: _language || language
                });

                if (response.ok === false) {
                    throw new Error("Failed to fetch translations");
                }
                // const translations = await response.json();
                setTranslations(response);
            } catch (error) {
                console.error("Error fetching translations:", error);
            }
        },
        500
    );

    const updateCachedTranslationData = useCallback(() => {
        if (typeof window !== "undefined") {
            const dataToCache = {
                translations,
                basicTranslations,
                country,
                language
            };
            try {
                sessionStorage.setItem(
                    "translationContext",
                    JSON.stringify(dataToCache)
                );
            } catch (error) {
                console.error(
                    "Failed to save translation context data to sessionStorage:",
                    error
                );
            }
        }
    }, [translations, basicTranslations, country, language]);

    useEffect(() => {
        updateCachedTranslationData();
    }, [updateCachedTranslationData]);

    const value = useMemo(
        () => ({
            translations,
            basicTranslations,
            country,
            language,
            setTranslations,
            setBasicTranslations,
            setCountry,
            setLanguage,
            fetchTranslations
        }),
        [translations, basicTranslations, country, language, fetchTranslations]
    );

    const providerValue = useMemo(
        () => ({
            ...value,
            ...testValues
        }),
        [value, testValues]
    );

    return (
        <TranslationContext.Provider value={providerValue}>
            <HtmlLangSync language={language} />
            {children}
        </TranslationContext.Provider>
    );
}

export const useTranslations = (): TranslationContextType =>
    useContext(TranslationContext);
