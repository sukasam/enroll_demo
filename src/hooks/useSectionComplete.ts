import { useSectionsConfig } from "Components/pages/Register/hooks";
import { useCountryConfig } from "Constants/countryConfig";
import { useOrder } from "Contexts/OrderContext";
import { useProducts } from "Contexts/ProductContext";
import { AddressType } from "Contexts/types/OrderContextTypes";
import { useUser } from "Contexts/UserContext";
import { useCallback, useMemo } from "react";

export type SectionCompleteType = {
    isAccountCreationComplete: boolean;
    isCatalogComplete: boolean;
    isShippingComplete: boolean;
    isDistributorAgreementComplete: boolean;
    isChangeHidden: (sectionId: number) => boolean;
};

export default function useSectionComplete(): SectionCompleteType {
    const { shoppingCart } = useProducts();

    const {
        signedTerms,
        userSignature,
        signatureDateTimeStamp,
        selectedPaymentMethod,
        shipToAddress
    } = useOrder();
    const { activeAccordionSection, isLoggedIn, userData, mainAddress } =
        useUser();
    const countryConfig = useCountryConfig();
    const sectionsConfig = useSectionsConfig(
        isLoggedIn,
        activeAccordionSection,
        countryConfig?.isNoPurchaseMarket
    );

    const isAccountCreationComplete = useMemo(
        () =>
            !!(
                isLoggedIn &&
                userData &&
                userData.email &&
                userData.firstName &&
                userData.lastName
            ),
        [isLoggedIn, userData]
    );

    const isCatalogComplete = useMemo(() => {
        const selectedProduct = shoppingCart?.find(product => product?.options);
        if (!selectedProduct) return false;
        if (selectedProduct.isDigital) return true;
        const unimateFlavor =
            selectedProduct?.options?.unimate_flavour?.title || null;
        const ballanceFlavor =
            selectedProduct?.options?.balance_pack?.title || null;
        return !!(unimateFlavor && ballanceFlavor);
    }, [shoppingCart]);

    const isShippingComplete = useMemo(() => {
        const address =
            countryConfig?.isNoPurchaseMarket ||
            countryConfig?.marketExceptions.onlyHasWillCall
                ? mainAddress
                : shipToAddress;

        // Define all possible fields and their requirements
        const fieldRequirements = {
            address1: true,
            city: !countryConfig?.marketExceptions.hiddenShippingQuotesCity,
            zip: !countryConfig?.marketExceptions.hiddenShippingQuotesZip,
            country: true,
            state: !countryConfig?.marketExceptions.hiddenShippingProvince
        };

        // Get required fields based on requirements
        const requiredFields = Object.entries(fieldRequirements)
            .filter(([_, isRequired]) => isRequired)
            .map(([field]) => field);

        return requiredFields.every(
            (field): field is keyof AddressType =>
                !!address?.[field as keyof AddressType]
        );
    }, [shipToAddress, mainAddress, countryConfig]);

    const isDistributorAgreementComplete = useMemo(
        () => !!(signedTerms && userSignature && signatureDateTimeStamp),
        [signedTerms, userSignature, signatureDateTimeStamp]
    );

    const isChangeHidden = useCallback(
        (sectionId: number) => {
            if (sectionId === activeAccordionSection) return true;
            const sectionConfig = sectionsConfig.find(
                section => section.id === sectionId
            );
            if (!sectionConfig) return true;
            switch (sectionConfig.type) {
                case "UpdateAccount":
                    return !isAccountCreationComplete;
                case "PackSection":
                    return !isCatalogComplete;
                case "DistributorOnly":
                    return !isCatalogComplete;
                case "ShippingSection":
                    return !isShippingComplete;
                case "DistributorAgreementsSection":
                    return !isDistributorAgreementComplete;
                case "PaymentSection":
                    return !selectedPaymentMethod;
                default:
                    return true;
            }
        },
        [
            sectionsConfig,
            activeAccordionSection,
            isAccountCreationComplete,
            isCatalogComplete,
            isShippingComplete,
            isDistributorAgreementComplete,
            selectedPaymentMethod
        ]
    );

    return {
        isAccountCreationComplete,
        isCatalogComplete,
        isShippingComplete,
        isDistributorAgreementComplete,
        isChangeHidden
    };
}
