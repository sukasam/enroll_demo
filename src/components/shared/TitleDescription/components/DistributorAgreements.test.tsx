import { render } from "@testing-library/react";
import { useOrder } from "Contexts/OrderContext";
import { useTranslations } from "Contexts/translation";
import useSectionComplete from "Hooks/useSectionComplete";
import DistributorAgreementsHooks from "./DistributorAgreements";

// Mock the necessary hooks and components
jest.mock("Contexts/OrderContext", () => ({
    useOrder: jest.fn()
}));

jest.mock("Contexts/translation", () => ({
    useTranslations: jest.fn()
}));

jest.mock("Hooks/useSectionComplete", () => jest.fn());

jest.mock(
    "Components/shared/Translate",
    () =>
        function Translate({
            children
        }: {
            children: React.ReactNode;
        }): React.ReactNode {
            return <span>{children}</span>;
        }
);

describe("DistributorAgreementsHooks", () => {
    beforeEach(() => {
        (useOrder as jest.Mock).mockReturnValue({
            signatureDateTimeStamp: null
        });

        (useTranslations as jest.Mock).mockReturnValue({
            country: "US",
            language: "en"
        });

        (useSectionComplete as jest.Mock).mockReturnValue({
            isDistributorAgreementComplete: false
        });

        global.Intl.DateTimeFormat = jest.fn().mockImplementation(() => ({
            format: (): string => "12/19/2023, 7:22 PM"
        })) as unknown as typeof Intl.DateTimeFormat;
        global.Intl.DateTimeFormat.supportedLocalesOf = jest
            .fn()
            .mockReturnValue(true);
    });

    it("renders correctly when agreement is not complete", () => {
        const { asFragment } = render(<DistributorAgreementsHooks />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders correctly when agreement is complete and has a valid signature date", () => {
        (useOrder as jest.Mock).mockReturnValue({
            signatureDateTimeStamp: "2023-12-19T19:22:00Z"
        });

        (useSectionComplete as jest.Mock).mockReturnValue({
            isDistributorAgreementComplete: true
        });

        const { asFragment } = render(<DistributorAgreementsHooks />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders null when there is no signature date", () => {
        (useOrder as jest.Mock).mockReturnValue({
            signatureDateTimeStamp: null
        });

        (useSectionComplete as jest.Mock).mockReturnValue({
            isDistributorAgreementComplete: true
        });

        const { asFragment } = render(<DistributorAgreementsHooks />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders null when agreement is not complete", () => {
        (useOrder as jest.Mock).mockReturnValue({
            signatureDateTimeStamp: "2023-12-19T19:22:00Z"
        });

        (useSectionComplete as jest.Mock).mockReturnValue({
            isDistributorAgreementComplete: false
        });

        const { asFragment } = render(<DistributorAgreementsHooks />);
        expect(asFragment()).toMatchSnapshot();
    });
});
