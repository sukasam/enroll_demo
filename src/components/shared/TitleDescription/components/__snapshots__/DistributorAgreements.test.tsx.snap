// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistributorAgreementsHooks renders correctly when agreement is complete and has a valid signature date 1`] = `
<DocumentFragment>
  <div
    data-testid="agreement_signed"
  >
    <span>
      agreement_signed
    </span>
    : 12/19/2023, 7:22 PM
  </div>
</DocumentFragment>
`;

exports[`DistributorAgreementsHooks renders correctly when agreement is not complete 1`] = `<DocumentFragment />`;

exports[`DistributorAgreementsHooks renders null when agreement is not complete 1`] = `<DocumentFragment />`;

exports[`DistributorAgreementsHooks renders null when there is no signature date 1`] = `<DocumentFragment />`;
